cmd/deployment/update.go

```diff
@@ -0,0 +1,179 @@
+      "ref_id": "my-apm-instance",
+      "region": "ece-region"
+    }
+  ]
+}
+#### Setting --prune-orphans, will cause any non-specified resources to be shut down.
+$ ecctl deployment update f44c06c3af6f85dac05023cf243f4ab1 -f deployment_example_update.json --prune-orphans
+setting --prune-orphans to "true" will cause any resources not specified in the update request to be removed from the deployment, do you want to continue? [y/n]: y
+{
+  "id": "f44c06c3af6f85dac05023cf243f4ab1",
+  "name": "my example cluster",
+  "resources": [
+    {
+      "id": "205745432f6345a4999dd2d77ceb1812",
+      "kind": "elasticsearch",
+      "ref_id": "my-es-cluster",
+      "region": "ece-region"
+    }
+  ],
+  "shutdown_resources": {
+    "apm": [
+      "90c3c9566f454861b0dc935c5c7420d8"
+    ],
+    "appsearch": [],
+    "elasticsearch": [],
+    "kibana": [
+      "3617594f01074b76a5ca4f903f9d33ec"
+    ]
+  }
+}`[1:]
+
+var updateCmd = &cobra.Command{
+	Use:     `update -f <file definition.json>`,
+	Short:   "Updates a deployment from a file definition, allowing certain flag overrides",
+	Long:    updateLong,
+	Example: updateExample,
+	PreRunE: cmdutil.MinimumNArgsAndUUID(1),
+	RunE: func(cmd *cobra.Command, args []string) error {
+		filename, _ := cmd.Flags().GetString("file")
+		var r models.DeploymentUpdateRequest
+		if err := cmdutil.DecodeFile(filename, &r); err != nil {
+			return err
+		}
+
+		pruneOrphans, _ := cmd.Flags().GetBool("prune-orphans")
+		r.PruneOrphans = &pruneOrphans
+
+		force, _ := cmd.Flags().GetBool("force")
+		var msg = `setting --prune-orphans to "true" will cause any resources not specified in the update request to be removed from the deployment, do you want to continue? [y/n]: `
+		if pruneOrphans && !force && !cmdutil.ConfirmAction(msg, os.Stderr, os.Stdout) {
+			return nil
+		}
+
+		skipSnapshot, _ := cmd.Flags().GetBool("skip-snapshot")
+		hidePrunedOrphans, _ := cmd.Flags().GetBool("hide-pruned-orphans")
+
+		var region string
+		if ecctl.Get().Config.Region == "" {
+			region = cmdutil.DefaultECERegion
+		}
+
+		res, err := deployment.Update(deployment.UpdateParams{
+			DeploymentID:      args[0],
+			API:               ecctl.Get().API,
+			Region:            region,
+			Request:           &r,
+			SkipSnapshot:      skipSnapshot,
+			HidePrunedOrphans: hidePrunedOrphans,
+		})
+
+		if err != nil {
+			return err
+		}
+
+		return ecctl.Get().Formatter.Format("", res)
+	},
+}
+
+func init() {
+	Command.AddCommand(updateCmd)
+	updateCmd.Flags().Bool("prune-orphans", false, "When set to true, it will remove any resources not specified in the update request, treating the json file contents as the authoritative deployment definition")
+	updateCmd.Flags().Bool("skip-snapshot", false, "Skips taking an Elasticsearch snapshot prior to shutting down the deployment")
+	updateCmd.Flags().Bool("hide-pruned-orphans", false, "Hides orphaned resources that were shut down (only relevant if --skip-orphan-prunning=true)")
```

Comment:
I don't think we have a `--skip-orphan-prunning` argument, do we? maybe `prune-orphans` instead?

Reply:
I'll update this flag's description